src/domain/safe/safe.repository.ts

```diff
@@ -323,4 +330,37 @@ export class SafeRepository implements ISafeRepository {
       ? null
       : this.multisigTransactionValidator.validate(page.results[0]);
   }
+
+  async proposeTransaction(
+    chainId: string,
+    safeAddress: string,
+    proposeTransactionDto: ProposeTransactionDto,
+  ): Promise<unknown> {
+    const transactionService =
+      await this.transactionApiManager.getTransactionApi(chainId);
+
+    const transaction = await transactionService.postMultisigTransaction(
+      safeAddress,
+      proposeTransactionDto,
+    );
+
+    await this.invalidateTransactions(chainId, safeAddress);
+    return transaction;
+  }
+
+  private async invalidateTransactions(
+    chainId: string,
+    safeAddress: string,
+  ): Promise<void> {
+    const cacheDirs = [
+      CacheRouter.getMultisigTransactionsCacheDir(chainId, safeAddress),
+      CacheRouter.getModuleTransactionsCacheDir(chainId, safeAddress),
+      CacheRouter.getTransfersCacheDir(chainId, safeAddress, false, false),
+      CacheRouter.getIncomingTransfersCacheDir(chainId, safeAddress),
+    ];
```

Comment:
Other `CacheDir` that we should invalidate:

- `getBalanceCacheDir` – the balance of the safe might get affected by this change
- `getSafeCacheDir` – the safe configuration might change (e.g. changing owners)
- `getCollectiblesCacheDir` – collectible collection might get updated
- `getAllTransactionsCacheDir` – similar to the ones that you already have

Reply:
Thank you for the detailed review! 🙂 I added those invalidations in [0b60e2e](https://github.com/safe-global/safe-client-gateway-nest/pull/292/commits/0b60e2e077ddd4ba2cec258d3903a52f5141487d)